sourcegraph/jetbrains
# Fix issues with chat colours after switching UI theme

It fixes [#229](https://github.com/sourcegraph/jetbrains/issues/229)

## Test plan

1. Ask Cody Chat question to which it will reply with various layout elements (list, code snippets, etc)
2. Change theme in settings or using `Themes` action.
3. Verify that chat text is readable, as well as that there is clear colour distinction between user and assistant sections. Both sections colors should also correspond to the chosen theme.

Repeat the above starting from different themes. 

## PR notes

We should try to avoid hardcoding color computations and instead always try to inherit them from the parent elements.
This way recompilations after theme change etc. are straightforward.
If we have to hardcode anything we must make sure it's in a place where it will be recomputed on either theme change or a repaint.

![Colors fix](https://github.com/sourcegraph/jetbrains/assets/1519649/10fcc661-d1e6-4c48-82ca-8cd8f133be43)
![Colors fix](https://github.com/sourcegraph/jetbrains/assets/1519649/480b3e26-5a40-49e0-8826-d4706aa8c170)
![Colors fix](https://github.com/sourcegraph/jetbrains/assets/1519649/7457b489-2d8f-4d23-a4ec-b8ea56655415)
